src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -768,6 +771,27 @@ object Parser {
         ghostPrimaryExp |
         operand
 
+    // TODO: change delimiter to { and implement required ambiguity resolution, current: !< X, ..., Z !>
+    // declaredPred!<d1, ..., dn!>
+    lazy val fpredConstruct: Parser[PPredConstructor] =
+      (idnUse ~ predConstructArgs) ^^ { p => PPredConstructor(PFPredBase(p._1), p._2) }
+
+    lazy val mpredConstruct: Parser[PPredConstructor] =
+      ???
+
+    lazy val predConstruct: Parser[PPredConstructor] =
+      fpredConstruct | mpredConstruct
+
+    lazy val predConstructArgs: Parser[Vector[Option[PExpression]]] =
+      ("!<" ~> (rep1sep(predConstructArg, ",") <~ ",".?).? <~ "!>") ^^ (opt => opt.getOrElse(Vector.empty))
+
+    lazy val predConstructArg: Parser[Option[PExpression]] =
+      (expression ^^ Some[PExpression]) | ("_" ^^^ None)
+
+    // TODO: add support for mpred, change fpredConstruct to predConstruct
+    lazy val predExprInstance: Parser[PExpression] =
+      fpredConstruct ~ callArguments ^^ PInvoke
```

Comment:
Can you not just delete this case? This should already be covered by the invoke parser.